content/4. Deeltaken/3. Processen/4. Implementatieproces/2. Bouwen softwaresysteem/While/1. While.md

```diff
@@ -0,0 +1,352 @@
+while ()  
+{  
+
+    // Kies een nieuw random getal  
+    number = rand.Next(0, 7);  
+}
+
+// Print het totaal
+```
+
+> [!info]- Mogelijke uitwerking
+> ``` csharp
+> using System;
+> 
+> // Variabel om het totaal bij te houden
+> int count = 0;
+>
+> // Kies een random getal
+> Random rand = new Random();
+> int number = rand.Next(0, 20);
+>
+> // While-lus om getallen te blijven vragen
+> while (number != 0)
+> {
+> 	 Console.WriteLine($"Nieuw getal is: {number}");
+> 	 count += number;  // Voeg het ingevoerde getal toe aan het totaal
+> 	 Console.WriteLine("Voer een positief getal in (0 om te stoppen): ");
+> 	 number = rand.Next(0, 20);
+>}
+>
+> // Print het totaal
+> Console.WriteLine($"Het totaal van alle ingevoerde getallen is: {count}");
+> ```
+
+---
+## Opdracht 4
+Schrijf een programma dat controleert of het juiste wachtwoord is ingevoerd.
+
+### Specificaties
+* Er wordt een random wachtwoord gekozen.
+* Het wachtwoord moet `C#Rocks` zijn.
+* Het programma blijft vragen totdat het juiste wachtwoord is ingevoerd.
+* Toon een succesbericht als het wachtwoord juist is
+
+### Verwachte output:
+Als invoer `123` is:
+```
+Voer het wachtwoord in:
+Fout wachtwoord probeer opnieuw.
+```
+Als invoer `C#Rocks` is:
+```
+Voer het wachtwoord in:
+Succes! Je bent ingelogd
+```
+
+### Nu jij:
+``` csharp runner
+using System;
+
+// Random wachtwoorden  
+string[] randomStrings = {"C#Rocks", "KotlinRocks", "JavaDoesntRock", "PythonRocks" };  
+  
+// Correcte wachtwoord  
+string password = "C#Rocks";  
+  
+// Random gekozen wachtwoord  
+string input = randomStrings[new Random().Next(randomStrings.Length)];  
+  
+// While-loop om het wachtwoord te blijven controleren.  
+while(){  
+
+	// Random gekozen wachtwoord is onjuist, kies een nieuwe
+    input = randomStrings[new Random().Next(randomStrings.Length)];  
+}
+
+// Print als wachtwoord correct is
+```
+
+> [!info]- Mogelijke uitwerking
+>``` csharp runner
+> using System;
+>
+> // Random wachtwoorden
+> string[] randomStrings = {"C#Rocks", "KotlinRocks", "JavaDoesntRock", "PythonRocks" };
+>        
+> // Correcte wachtwoord
+> string password = "C#Rocks";
+>        
+> // Random gekozen wachtwoord
+> string input = randomStrings[new Random().Next(randomStrings.Length)];
+>        
+> // While-loop om het wachtwoord te blijven controleren.
+> while(input != password)
+> {
+> 	Console.WriteLine($"Gekozen wachtwoord is: {input}");
+> 	Console.WriteLine("Fout wachtwoord probeer opnieuw.");
+> 	input = randomStrings[new Random().Next(randomStrings.Length)];
+> }
+>   
+> // Print als wachtwoord correct is
+> Console.WriteLine($"Gekozen wachtwoord is: {input}");
+> Console.WriteLine("Succes! Je bent ingelogd");
+>```
+
+---
+## Opdracht 5
+Schrijf een programma dat een random waarde van een dobbelsteen (1 t/m 6) kiest. Het programma moet een nieuwe worp blijven doen zolang de waarde van de dobbelsteen niet 6 is. Wanneer 6 gegooid is moet er getoond worden hoeveel worpen er gedaan zijn.
```

Comment:
Opmerking: Leuke opgave, deze. Is het handig om hier nog ergens een link te maken naar scoping? Gezien op deze pagina nog niet eerder is gesproken over variable/value permanence.

Reply:
Nice, er was in eerste instantie geen plan om voor scoping OI/PI/Deeltaken te maken, gaan we nu toevoegen aan de lijst :)